### ISSUE TITLE
Refactor to use new core bundle classes functionality

### ISSUE THREAD

**Issue by DieterHolvoet** [wieni/wmmodel#16]
## Description

We'll have to increase the minimum core version, but we'll be able to get rid of the patch. See https://www.drupal.org/project/drupal/issues/2570593.

**Comment by DieterHolvoet** [comment]
I started a WIP pull request. Some remarks:

- We should consider removing `Drupal\wmmodel\Factory\ModelFactory(Interface)` since all functionality is now possible using core services. We could also keep it as a utility class.
- We should consider making a core issue for `WmModel::loadMultiple`. Calling `loadMultiple` on a bundle class should probably filter the returned entities by bundle, as it does in our trait. Core does the same when passing an array of entities to `postLoad`. If this is integrated in core, we can remove `Drupal\wmmodel\Entity\Traits\WmModel`.
- We should consider removing `Drupal\wmmodel\Entity\Interfaces\WmModelInterface` since it has no added value.

**Comment by RobinHoutevelts** [comment]
> We should consider removing Drupal\wmmodel\Factory\ModelFactory(Interface) since all functionality is now possible using core services. We could also keep it as a utility class.

The less custom code, the better no? I'd remove it if we can achieve the goal of this module without the ModelFactory.

> We should consider making a core issue for WmModel::loadMultiple. Calling loadMultiple on a bundle class should probably filter the returned entities by bundle, as it does in our trait. Core does the same when passing an array of entities to postLoad. If this is integrated in core, we can remove Drupal\wmmodel\Entity\Traits\WmModel.

I'm no fan of the static `::load*` family. You should inject the `EntityTypeManager` or the relevant `EntityStorage` and load from there. If this is the only reason we currently have the trait, we should definitely remove it. 👍

However the trait has been in **all** our projects the past 5 years. So perhaps we should add a `Rector` to facilitate an upgrade. Did we ever do that before?

> We should consider removing Drupal\wmmodel\Entity\Interfaces\WmModelInterface since it has no added value.

I agree 👍 However we have been (ab)using this for the past 5 years to identify entities that aren't part of any core or contrib modules.

I'm not sure we can just search/replace it in our projects with the `EntityInterface`

Edit: When I search our SourceGraph, it looks like I'm exaggerating 🤔 

```
WmModelInterface -file:src\/Entity\/ -file:settings\.php
```

**Comment by DieterHolvoet** [comment]
I created a core issue for the `loadMultiple` behaviour, let's see what they say: https://www.drupal.org/project/drupal/issues/3252421.

I agree it's bad practice to use it, but the methods are there and I do believe that they should work differently when called on bundle classes.

**Comment by DieterHolvoet** [comment]
Discussion is still ongoing in that issue, opinions are divided. I propose we get rid of those methods, it's not up to us to fix that issue and it's not that important as well.

**Comment by DieterHolvoet** [comment]
@RobinHoutevelts while we're at it, I think we should also remove this line:
```php
is_a($argument->getType(), FormStateInterface::class, true)
```
from `ModelValueResolver`. I'm not against what it does, I just think wmmodel isn't the right place for that. I think a core patch for [this issue](https://www.drupal.org/project/drupal/issues/3006502) in the relevant projects would be better.

**Comment by RobinHoutevelts** [comment]
> Discussion is still ongoing in that issue, opinions are divided. I propose we get rid of those methods, it's not up to us to fix that issue and it's not that important as well.

I've gone over our projects with SourceGraph and we don't seem to rely on the `::load*` methods very often. So I don't mind dropping support for those 👍

**Comment by DieterHolvoet** [comment]
Finished and merged! Once we're done with #18 and #19 we can release.
